m4/ct_check_tls.m4

```diff
@@ -0,0 +1,37 @@
+dnl Checks for thread-local storage support.
+dnl
+dnl Taken from the openvswitch config code (Apache 2.0 License)
+dnl with some local modifications. Does not include <threads.h>
+dnl as this does not currently exist on GCC.
+dnl Checks whether the compiler and linker support the C11
+dnl thread_local macro from <threads.h>, and if so defines
+dnl HAVE_THREAD_LOCAL.  If not, checks whether the compiler and linker
+dnl support the GCC __thread extension, and if so defines
+dnl HAVE___THREAD.
+AC_DEFUN([CT_CHECK_TLS],
+  [AC_CACHE_CHECK(
+     [whether $CC has <threads.h> that supports thread_local],
+     [ct_cv_thread_local],
+     [AC_LINK_IFELSE(
```

Comment:
This *might* work as a `AC_COMPILE_IFELSE` instead of an `AC_LINK_IFELSE`, but if it works like this, I'd be tempted to leave it well alone. :stuck_out_tongue: